collector/src/test/java/com/google/cloud/trace/zipkin/ZipkinCollectorIntegrationTest.java

```diff
@@ -0,0 +1,150 @@
+    webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT,
+    properties = {ZIPKIN_CONFIG_NAMES, "zipkin.storage.stackdriver.project-id=test_project",
+            "zipkin.storage.stackdriver.api-host=localhost:${mock.stackdriver.port}"})
+@ContextConfiguration(initializers = ZipkinCollectorIntegrationTest.RandomPortInitializer.class)
+public class ZipkinCollectorIntegrationTest
+{
+  //LOTS_OF_SPANS is too much! I need at most 30!
+  private static final List<Span> TEST_SPANS = unmodifiableList(asList(Arrays.copyOfRange(LOTS_OF_SPANS, 0, min(30, LOTS_OF_SPANS.length))));
+
+  @Autowired
+  private StackdriverMockServer mockServer;
+
+  @Autowired
+  private TestRestTemplate restTemplate;
+
+  @After
+  public void cleanup()
+  {
+    this.mockServer.reset();
+  }
+
+  @Test
+  public void traceSinkGetsCalled() throws InterruptedException
+  {
+    final Set<Long> spanIds = toIds(TEST_SPANS);
+    assertThat("Expected to test at least one span", spanIds, hasSize(greaterThan(0)));
+    assertThat("Unexpected trace in Stackdriver", mockServer.spanIds(), Matchers.<Long>empty());
+    final CountDownLatch spanCountdown = new CountDownLatch(spanIds.size());
+    mockServer.setSpanCountdown(spanCountdown);
+
+    for (List<Span> partition : Lists.partition(TEST_SPANS, 3))
+    {
+      this.restTemplate.postForLocation("/api/v1/spans", Codec.JSON.writeSpans(partition));
+    }
+    spanCountdown.await(1, TimeUnit.SECONDS);
+
+    assertThat(spanCountdown.getCount(), equalTo(0l));
+    assertThat("Not all spans made it to Stackdriver", spanIds, equalTo(mockServer.spanIds()));
+  }
+
+  private Set<Long> toIds(List<Span> spans)
+  {
+    final Set<Long> spanIds = new HashSet<>();
+    for (Span span : spans)
+    {
+      spanIds.add(span.id);
+    }
+    return spanIds;
+  }
+
+  @Configuration
+  public static class TestConfiguration
+  {
+    @Bean(name = "googleCredentials")
+    @Primary
+    public Credentials mockGoogleCredentials() throws IOException
+    {
+      return mock(GoogleCredentials.class);
+    }
+
+    @Bean(name = "traceSink")
+    @Primary
+    TraceSink traceSink(Credentials credentials, ZipkinStackdriverStorageProperties storageProperties)
+        throws IOException, NoSuchFieldException, IllegalAccessException
+    {
+      final GrpcTraceSink traceSink = new GrpcTraceSink(storageProperties.getApiHost(), credentials);
+      final ManagedChannel managedChannel = ManagedChannelBuilder.forTarget(storageProperties.getApiHost()).usePlaintext(true).build();
+      TraceServiceGrpc.TraceServiceBlockingStub traceService = TraceServiceGrpc.newBlockingStub(managedChannel)
+          .withCallCredentials(MoreCallCredentials.from(credentials));
+
+      final Field traceServiceField = GrpcTraceSink.class.getDeclaredField("traceService");
+      traceServiceField.setAccessible(true);
+      traceServiceField.set(traceSink, traceService);
```

Comment:
This is a bit gross, but we'll be updating the SDK shortly so we can get rid of this. :)